ssl/t1_lib.c

```diff
@@ -133,7 +133,8 @@ int ssl_check_serverhello_tlsext(SSL *s);
 #endif
 
 #define CHECKLEN(curr, val, limit) \
-    (((curr) >= (limit)) || (size_t)((limit) - (curr)) < (size_t)(val))
+    (((size_t)(curr) >= (size_t)(limit)) || \
+     ((size_t)(limit) - (size_t)(curr)) < (size_t)(val))
```

Comment:
Casting pointers to a size_t is wrong, but happens to work on most platforms. If you need to cast it, it needs to be an uintptr_t. But I currently don't see what is wrong with the old macro.

Note that that code is also still present in 1.1.0, but was removed in 1.1.1.